coreneuron/io/reports/report_event.cpp

```diff
@@ -88,8 +90,11 @@ void ReportEvent::deliver(double t, NetCvode* nc, NrnThread* nt) {
                                 gids_to_report.data(),
                                 report_path.data());
 #endif
-        send(t + dt, nc, nt);
+        // Deterministic event time per report to avoid deadlocks
+        send(t + dt + report_t_shift_, nc, nt);
```

Comment:
I guess that's the only way to do it without changing the API of reportinglib? it's a bit ugly that we need to add this artificial `1e-6*report_idx` to fix the order.